Source/Controllers/Preferences/Panes/SPNetworkPreferencePane.m

```diff
@@ -263,11 +227,13 @@ - (void)updateSSHConfigPopUp
 		
 		// every bookmark is saved in relation to it's abslute path
 		while (key = [keyEnumerator nextObject]) {
-			NSString *itemTitle = [key substringFromIndex:[@"file://" length]];
-			
-			[sshConfigChooser addItemWithTitle:itemTitle];
-			
-			count++;
+            
+            if([key hasPrefixWithPrefix:@"file://" caseSensitive:YES] == YES){
+                // save the filename without the file protocol
+                NSString *itemTitle = [key substringFromIndex:[@"file://" length]];
+                [sshConfigChooser addItemWithTitle:itemTitle];
+                count++;
+            }
```

Comment:
Should we log or something if somehow it doesn't have file://
Or, a better way to structure this block functionally would be an early return, rather than a guarded if. ex:

```
while (key = [keyEnumerator nextObject]) {
    if([key hasPrefixWithPrefix:@"file://" caseSensitive:YES] != YES){
        continue;
    }
    NSString *itemTitle = [key substringFromIndex:[@"file://" length]];
    [sshConfigChooser addItemWithTitle:itemTitle];
    count++;
}
```

This way the normal, healthy flow of code flows from the beginning to the end of the method/block and the uncommon guard condition is treated as an early-break pre-condition. This allows our core working code to be nested one level less deep and improves the long-term legibility of the code.

Reply:
Ok, not my code ... but will take a look :)

Reply:
sorted.